corehq/apps/hqcase/views.py

```diff
@@ -193,10 +193,42 @@ def _handle_list_view(request):
 
 
 def _handle_ext_put(request, external_id):
-    pass  # TODO: ...
-
+    try:
+        data = json.loads(request.body.decode('utf-8'))
+    except (UnicodeDecodeError, json.JSONDecodeError):
+        return JsonResponse({'error': "Payload must be valid JSON"}, status=400)
+    if 'external_id' not in data:
+        data['external_id'] = external_id
 
-def _handle_case_update(request, is_creation, case_id=None):
+    try:
+        bulk_fetch_results_dict = get_bulk(
+            request.domain,
+            request.couch_user,
+            case_ids=[],
+            external_ids=[external_id],
+        )
+    except UserError as err:
+        return JsonResponse({'error': str(err)}, status=400)
+    case = bulk_fetch_results_dict['cases'][0]
+    if case.get('error') == 'not found':
+        is_creation = True
+    else:
+        is_creation = False
```

Comment:
I didn't think anything of it with the `_handle_ext_get` function, but here I'm wondering if this wouldn't be better served with something bespoke, either extracted from `get_bulk`'s internals or paralleling it.  There's a lot that goes on there that doesn't apply here, and it makes this code a bit weird - looks like it can raise a `UserError`, but at present that's only for page size issues.  Reading the error message from an API call seems like a convoluted way to go about seeing if a case is present, and what if there's another error returned?

Instead, you could just do a simple get by external ID (failing if there are multiple matches) and run it past `user_can_access_case` for location safety.  That could be a util used here and in `_handle_ext_get`.